src/main/java/it/pn/frontend/e2e/pages/mittente/DisserviziAppPAPage.java

```diff
@@ -41,22 +51,44 @@ public void waitLoadStatoDellaPiattaformaPage() {
             getWebDriverWait(3).until(ExpectedConditions.textToBe(disserviziPageSubTitle, "Verifica il funzionamento di SEND, visualizza lo storico dei disservizi e scarica le relative attestazioni opponibili a terzi."));
             getWebDriverWait(10).withMessage("Non si visualizza correttamente la sezione disservizi")
                     .until(ExpectedConditions.visibilityOfElementLocated(disserviziBoxAlert));
+
             if (this.element(disserviziBoxAlert).getText().contains("C'è un disservizio in corso")) {
                 getWebDriverWait(3).until(ExpectedConditions.textToBe(disserviziBoxAlert, "C'è un disservizio in corso. Per maggiori dettagli, consulta la tabella qui sotto."));
             } else {
                 getWebDriverWait(3).until(ExpectedConditions.textToBe(disserviziBoxAlert, "Tutti i servizi di SEND sono operativi."));
             }
+
             getWebDriverWait(10).withMessage("Non si visualizza correttamente l'ultimo aggiornamento della pagina")
                     .until(ExpectedConditions.visibilityOfElementLocated(disserviziLastUpdate));
             getWebDriverWait(10).withMessage("Non si visualizza correttamente la tabella dei disservizi")
                     .until(ExpectedConditions.visibilityOf(disserviziTable));
+            getWebDriverWait(10).withMessage("Non si visualizza correttamente il titolo della tabella dei disservizi")
+                    .until(ExpectedConditions.visibilityOfElementLocated(disserviziTitleOfTable));
+
             logger.info("Si visualizza correttamente la sezione disservizi");
         } catch (TimeoutException e) {
             logger.error("Non si visualizza correttamente la sezione disservizi con errore:" + e.getMessage());
             Assert.fail("Non si visualizza correttamente la sezione disservizi con errore" + e.getMessage());
         }
     }
 
+    public void getDateDisservice() {
+
+        List<WebElement> disserviziTableRows = disserviziTable.findElements(By.id("tableDowntimeLog.row"));
+        if (!disserviziTableRows.isEmpty()) {
+            WebElement primaRiga = disserviziTableRows.get(0);
+            String dataInizioPrimaRiga = primaRiga.findElements(By.xpath("//td[@data-testid='tableDowntimeLog.row.cell']//div//div//p[contains(text(), 'ore')]")).get(0).getText();
+            String dataFinePrimaRiga = primaRiga.findElements(By.xpath("//td[@data-testid='tableDowntimeLog.row.cell']//div//div//p[contains(text(), 'ore')]")).get(1).getText();
+            disservice.setDataDa(dataInizioPrimaRiga);
+            disservice.setDataA(dataFinePrimaRiga);
+
+        } else {
+            logger.error("non é stato possibile recuperare i dati dalla tabella dei disservizi");
+            Assert.fail("non é stato possibile recuperare i dati dalla tabella dei disservizi");
+
+        }
+    }
+
```

Comment:
I worked on this a couple of months ago, so maybe you can use the existing one

Reply:
you created this metod waitLoadDisserviziTable but I would still have to create a method to get the dates explicitly

Reply:
check this method and see if it can help with what you're trying to accomplish:

https://github.com/pagopa/pn-frontend-e2e/blob/fec4568e6f065463fd437e820f32394b0c3feabf/src/main/java/it/pn/frontend/e2e/pages/mittente/DisserviziAppPAPage.java#L115-L136


As you can see there is a check for the "date" field, maybe you can just update this one and make it more specific.

Reply:
this method control if disservis is in progress

i need to control if is completed